pkg/labels/v2/label_ext.go

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2
+
+import (
+	"bytes"
+	"fmt"
+	"net/netip"
+	"strings"
+)
+
+//
+// Convenience methods for 'Label'
+//
+
+const (
+	// LabelSourceUnspec is a label with unspecified source
+	LabelSourceUnspec = "unspec"
+
+	// LabelSourceAny is a label that matches any source
+	LabelSourceAny = "any"
+
+	// LabelSourceAnyKeyPrefix is prefix of a "any" label
+	LabelSourceAnyKeyPrefix = LabelSourceAny + "."
+
+	// LabelSourceK8s is a label imported from Kubernetes
+	LabelSourceK8s = "k8s"
+
+	// LabelSourceK8sKeyPrefix is prefix of a Kubernetes label
+	LabelSourceK8sKeyPrefix = LabelSourceK8s + "."
+
+	// LabelSourceContainer is a label imported from the container runtime
+	LabelSourceContainer = "container"
+
+	// LabelSourceCNI is a label imported from the CNI plugin
+	LabelSourceCNI = "cni"
+
+	// LabelSourceReserved is the label source for reserved types.
+	LabelSourceReserved = "reserved"
+
+	// LabelSourceCIDR is the label source for generated CIDRs.
+	LabelSourceCIDR = "cidr"
+
+	// LabelSourceCIDRGroup is the label source used for labels from CIDRGroups
+	LabelSourceCIDRGroup = "cidrgroup"
+
+	// LabelSourceNode is the label source for remote-nodes.
+	LabelSourceNode = "node"
+
+	// LabelSourceFQDN is the label source for IPs resolved by fqdn lookups
+	LabelSourceFQDN = "fqdn"
+
+	// LabelSourceReservedKeyPrefix is the prefix of a reserved label
+	LabelSourceReservedKeyPrefix = LabelSourceReserved + "."
+
+	// LabelSourceDirectory is the label source for policies read from files
+	LabelSourceDirectory = "directory"
+)
+
+// EqualIgnoringAnySource returns true if source (ignoring 'any'), Key and Value are equal and false otherwise.
```

Comment:
Is this useful? Label comparison with any source is [not commutative](https://github.com/cilium/cilium/blob/77b81c0df6bd50cdfb30e17c49864cc9e3f9bb9d/pkg/labels/labels.go#L376); if we are matching on any bidirectionally it may indicate a bug.